buildSrc/src/main/kotlin/otel.java-library-conventions.gradle.kts

```diff
@@ -1,58 +0,0 @@
-import org.jetbrains.kotlin.gradle.dsl.JvmTarget
```

Comment:
Oof, I guess this was cruft just left over from a previous rename?

Reply:
The convention configs for animalsniffer checks on plain Java libs are slightly different than for Android libs, for a moment I thought about adding all that config in a separate convention file and so on, but then I realized that the plain Java lib conventions aren't used anywhere right now, so it seems like it's better to just remove them instead of making this PR bigger by adding all that other config that's not needed right now.